runner_manager/models/runner_group.py

```diff
@@ -200,6 +214,10 @@ def need_new_runner(self) -> bool:
     def create_github_group(self, github: GitHub) -> GitHubRunnerGroup:
         """Create a GitHub runner group."""
 
+        exists = self.github_group_exists(github)
+        if exists is not None and self.id is None:
+            self.id = exists.id
+
```

Comment:
Please forgive me for my previous suggestion, I think I was tired, this is more towards what I was trying to achieve:
```suggestion
        if self.id is None:
            exists = self.github_group_exists(github)
            self.id = exists.id if exists is not None else None

```

There's most likely a prettier/better way to do it, all I'm trying to achieve here is to not look for the group if we already have `self.id` defined.